File: source/utility/Timer.java
package utility;

public class Timer {
    private long elapsedStart;
    private long deltaStart;
    private long deltaEnd;

    public Timer() {
        long currTime = System.nanoTime();
        elapsedStart = currTime;
        deltaStart = currTime;
        deltaEnd = currTime;
    }

    public void reset() {
        updateElapsedT();
        updateDeltaT();
    }

    public void updateElapsedT() {
        elapsedStart = System.nanoTime();
    }
    public float getElapsedT() {
        return (System.nanoTime() - elapsedStart) * 1e-9f;
    }

    public void updateDeltaT() {
        deltaStart = deltaEnd;
        deltaEnd = System.nanoTime();
    }
    public float getDeltaT() {
        return (deltaEnd - deltaStart) * 1e-9f;
    }
}

File: source/utility/Timer.kt
package utility

class Timer {
    private var elapsedStart: Long
    private var deltaStart: Long
    private var deltaEnd: Long

    init {
        val currTime = System.nanoTime()
        elapsedStart = currTime
        deltaStart = currTime
        deltaEnd = currTime
    }

    fun reset() {
        updateElapsedT()
        updateDeltaT()
    }

    fun updateElapsedT() {
        elapsedStart = System.nanoTime()
    }

    fun getElapsedT(): Float {
        return (System.nanoTime() - elapsedStart) * 1e-9f
    }

    fun updateDeltaT() {
        deltaStart = deltaEnd
        deltaEnd = System.nanoTime()
    }

    fun getDeltaT(): Float {
        return (deltaEnd - deltaStart) * 1e-9f
    }
}